docker/Dockerfile

```diff
@@ -1,43 +1,80 @@
-FROM nvidia/cuda:12.2.2-devel-ubuntu22.04
+ARG UBUNTU_VERSION=22.04
+ARG NVIDIA_CUDA_VERSION=12.3.1
+
+#
+# Docker builder stage.
+#
+FROM nvidia/cuda:${NVIDIA_CUDA_VERSION}-devel-ubuntu${UBUNTU_VERSION} as builder
 
 ARG COLMAP_GIT_COMMIT=main
-ARG CUDA_ARCHITECTURES=native
+ARG CUDA_ARCHITECTURES=50
```

Comment:
Should I leave `ARG CUDA_ARCHITECTURES=native`? It didn't work when I run it myself without the calls from the yaml file.

Reply:
I have addressed this comment in PR https://github.com/colmap/colmap/pull/2356